### PR TITLE
feat(FR-2269): add E2E test for login after self-service signup

### LINKED ISSUE #5908: Add E2E test for login after self-service signup
## Summary

Add an E2E (Playwright) test that exercises the complete self-service signup flow and verifies that a newly signed-up account can subsequently log in. This covers the path from the signup modal through the `/auth/signup` API call to a successful post-signup login.

## Background

The existing E2E test suite (`e2e/auth/login.spec.ts`, `e2e/user/user-crud.spec.ts`) covers admin-created user login and various login edge cases, but does not cover the **self-service signup** path. The `SignupModal` component (`react/src/components/SignupModal.tsx`) posts to `/auth/signup` using an anonymous client. After signup, the user is expected to be able to log in immediately when `allowSignupWithoutConfirmation` is `true`, or after email verification when it is `false`.

The Jira task FR-2269 (subtask of FR-2258 / FR-2263) explicitly calls for test coverage of this gap.

## Proposed Solution

Create `e2e/auth/signup.spec.ts` as a new Playwright spec under the `@auth` tag group. The test should:

1. Enable the signup link in `config.toml` by setting `signup_support = true` and `allowSignupWithoutConfirmation = true` via `modifyConfigToml`.
2. Navigate to the login page and open the signup modal via the "Create a new account" link.
3. Fill the signup form (email, username, password, password confirmation, Terms of Service checkbox).
4. Submit the form and assert that signup succeeds (modal closes without an error toast).
5. Log in with the newly registered credentials and assert successful navigation to `/start`.
6. Clean up the test account via the admin API after each test run (deactivate → purge, following the pattern used in `e2e/user/user-crud.spec.ts`).

An additional negative test should verify that signing up with an already-registered email shows the appropriate error message.

## Implementation Notes

- Use `modifyConfigToml` from `e2e/utils/test-util.ts` to toggle `signup_support` and `allowSignupWithoutConfirmation` on/off per test.
- The signup modal form fields are labelled `E-mail`, `Username`, `Password`, and `Password (again)`. The TOS checkbox must be checked before submitting.
- The backend signup endpoint is `POST /auth/signup` (see `SignupModal.tsx` line 64).
- When `allowSignupWithoutConfirmation` is `true`, signup creates the account immediately without an email verification step, so the test can proceed directly to login.
- When `allowSignupWithoutConfirmation` is `false`, the modal shows a "Thank you / email sent" dialog. This path requires backend email delivery and should be covered by a separate, `test.fixme`-marked test until email can be mocked or a test SMTP server is available.
- Use `test.describe.serial` to ensure cleanup order is deterministic.
- Cleanup should use `loginAsAdmin`, navigate to `credential`, deactivate, then purge — identical to the pattern in `e2e/user/user-crud.spec.ts`.
- Tag new tests with `@smoke`, `@auth`, and `@functional`.

## Acceptance Criteria

- [ ] `e2e/auth/signup.spec.ts` is created with at minimum:
  - `user can sign up and immediately log in when allowSignupWithoutConfirmation is true`
  - `signup with an already-registered email shows an error`
- [ ] Tests use `modifyConfigToml` to enable signup support
- [ ] Test account is cleaned up after each run via admin deactivate + purge
- [ ] New spec passes locally against a live Backend.AI cluster
- [ ] Tests are tagged `@smoke`, `@auth`, `@functional`
- [ ] A `test.fixme` stub exists for the email-confirmation signup path

---

### Title: Add test code to log in with a new account after signing up

Add E2E or integration test code to verify login with a newly signed-up account.

Originally from: FR-2263 (subtask of FR-2258)

JIRA Issue: FR-2269

### PR DESCRIPTION
**Pull request by agatha197**
Resolves #5908 ([FR-2269](https://lablup.atlassian.net/browse/FR-2269), [FR-2263](https://lablup.atlassian.net/browse/FR-2263))

## Summary
- Add `e2e/auth/signup.spec.ts` with E2E tests for self-service signup flow
- Test: user can open signup modal, fill form, and sign up successfully
- Test: duplicate email signup shows error
- `test.fixme`: login after signup (blocked by FR-2268 backend bug)
- `test.fixme`: email-confirmation path (needs test SMTP server)
- Cleanup: admin deactivates and purges test account in afterAll
- Tagged `@smoke`, `@auth`, `@functional`

## Files Changed
- `e2e/auth/signup.spec.ts` (new)

[FR-2269]: https://lablup.atlassian.net/browse/FR-2269
[FR-2263]: https://lablup.atlassian.net/browse/FR-2263

### CODE REVIEW

File: e2e/auth/signup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      // Enable signup in config.toml and allow immediate login without email confirmation
+      await modifyConfigToml(page, request, {
+        general: {
+          connectionMode: 'SESSION',
+          apiEndpoint: '',
+          apiEndpointText: '',
+          signupSupport: true,
+          allowSignupWithoutConfirmation: true,
+        },
+      });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Set `apiEndpoint` and `apiEndpointText` to `webServerEndpoint` in `beforeEach` so `showSignupDialog()` resolves a non-empty endpoint and opens the signup modal correctly.

File: e2e/auth/signup.spec.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          if (overlay) overlay.remove();
+        })
+        .catch(() => {});
+    });
+
+    test('user can open the signup modal and sign up successfully', async ({
+      page,
+    }) => {
+      // The "Sign up" link should be visible when signup_support is true
+      const signUpLink = page.getByRole('button', { name: 'Sign up' });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Changed to `getByRole('link', { name: 'Sign up' })` to match the `Typography.Link` rendering.

File: e2e/auth/signup.spec.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+      // Check the Terms of Service / Privacy Policy agreement checkbox
+      await modal.getByRole('checkbox').click();
+
+      // Submit the form via the primary "Signup" button inside the modal footer
+      await modal.getByRole('button', { name: 'Signup' }).click();
+
+      // When allowSignupWithoutConfirmation is true the modal closes without an
+      // error toast — verify it disappears within a reasonable timeout
+      await expect(modal).toBeHidden({ timeout: 15_000 });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Replaced `.ant-modal` with `page.getByRole('dialog', { name: /Signup/i })` to target the signup dialog specifically and avoid matching the login modal.

File: e2e/auth/signup.spec.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        page.getByTestId('webui-breadcrumb').getByText('Start'),
+      ).toBeVisible();
+    });
+
+    test('signup with an already-registered email shows an error', async ({
+      page,
+    }) => {
+      const signUpLink = page.getByRole('button', { name: 'Sign up' });
+      await expect(signUpLink).toBeVisible({ timeout: 10_000 });
+      await signUpLink.click();
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Changed to `getByRole('link', { name: 'Sign up' })`.

File: e2e/auth/signup.spec.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+      const modal = page.locator('.ant-modal');
+      await expect(modal).toBeVisible({ timeout: 5_000 });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Replaced `.ant-modal` with `page.getByRole('dialog', { name: /Signup/i })`.

File: e2e/auth/signup.spec.ts [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+      // Override config to require email confirmation
+      await modifyConfigToml(page, request, {
+        general: {
+          signupSupport: true,
+          allowSignupWithoutConfirmation: false,
+        },
+      });
+      await page.goto(webuiEndpoint);
+
+      const signUpLink = page.getByRole('button', { name: 'Sign up' });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts]: Fixed. Updated to `getByRole('link', { name: 'Sign up' })` in the fixme stub as well.

File: e2e/auth/signup.spec.ts [hunk 7]
```diff
@@ -0,0 +1,221 @@
+// spec: e2e/auth/signup.spec.ts
+// Covers: FR-2269 — Add E2E test for login after self-service signup
+import { PurgeUsersModal } from '../utils/classes/user/PurgeUsersModal';
```
[the remark this answers is not in the document]
Comment (agatha197) [reply e2e/auth/signup.spec.ts:3]: Fixed. Updated `E2E_COVERAGE_REPORT.md` to include the new signup spec with its test cases in the Authentication section and refreshed the last-updated date.

**Review by agatha197 — COMMENTED** [review]
Not planned.
